src/main/java/com/google/neighborgood/helper/TaskGroup.java

```diff
@@ -19,31 +19,31 @@
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.EntityNotFoundException;
 import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.memcache.ErrorHandlers;
+import com.google.appengine.api.memcache.MemcacheService;
+import com.google.appengine.api.memcache.MemcacheServiceFactory;
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
 import com.google.neighborgood.data.Task;
-import com.google.neighborgood.data.User;
 import java.util.ArrayList;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
+import java.util.logging.Level;
 
 /**
- * Helper class that stores the HTML depiction of tasks in groups of 10 or less along with whether
- * or not the end of the query has been reached
+ * Helper class that stores tasks in groups of 10 or less along with whether or not the end of the
+ * query has been reached
  */
 public class TaskGroup {
-  private static Map<String, User>
-      usersInfo; // Stores task owner's user info to prevent querying multiple times in
-  // datastore for the same user's info
+  private static MemcacheService syncCache;
   private static DatastoreService datastore;
   private final boolean userLoggedIn;
   private int currentTaskCount;
   private boolean endOfQuery;
   private List<Task> tasks;
 
   public TaskGroup() {
-    this.usersInfo = new HashMap<String, User>();
+    this.syncCache = MemcacheServiceFactory.getMemcacheService();
+    syncCache.setErrorHandler(ErrorHandlers.getConsistentLogAndContinue(Level.INFO));
```

Comment:
If there is an error with the syncCache, what will the error handler do?

Reply:
This essentially just logs the error and emulates a cache-miss behavior (so it would just return null when an error arises when using get() )

Reply:
So when the cache doesn't work, we fail to load the user's nickname? Should we have a backup plan for that? I don't think you need to really implement that, but maybe you would like to mention it in the design doc.

Reply:
That's why if the cache value returns null, the nickname is retrieved from datastore instead and pushed into the cache for future use :)

Reply:
Ah you change it back to retrieving from datastore. I thought you would keep the hashmap with the cache and when the cache doesn't work we would use the hashmap. And then I found you remove the hashmap and I thought you didn't have a backup plan for that. My fault. Then it looks pretty good to me.